app/views/base.py

```diff
@@ -251,11 +254,23 @@ def build_absolute_path(self, path=None, repo=None):
         if repo:
             path = '/%s%s' % (repo, path)
         if self._request_is_for_prefixed_path():
-            return '/%s%s' % (site_settings.OPTIONAL_PATH_PREFIX, path)
+            res = '/%s%s' % (site_settings.OPTIONAL_PATH_PREFIX, path)
         else:
-            return path
-
-    def build_absolute_uri(self, path=None, repo=None):
+            res = path
+        if params:
+            url_parts = list(urlparse.urlparse(res))
+            url_params = dict(urlparse.parse_qsl(url_parts[4]))
```

Comment:
is it worth documenting the number 4?

Reply:
IMO the context of assigning it to the `url_params` variable is enough.